core/src/main/java/org/apache/calcite/sql/SqlIntervalQualifier.java

```diff
@@ -1172,6 +1208,9 @@ public int[] evaluateIntervalLiteral(String value, SqlParserPos pos,
     case YEAR_TO_MONTH:
       return evaluateIntervalLiteralAsYearToMonth(typeSystem, sign, value,
           value0, pos);
+    case QUARTER:
+      return evaluateIntervalLiteralAsQuarter(typeSystem, sign, value, value0,
+        pos);
```

Comment:
Notice how this section, which is the only one that was required to implement this, uses `TimeUnit.QUARTER` and does not use `INTERVAL_QUARTER` at all.